openshift-kube-apiserver/admission/customresourcevalidation/node/validate_node_config.go

```diff
@@ -57,7 +78,11 @@ func toConfigNodeV1(uncastObj runtime.Object) (*configv1.Node, field.ErrorList)
 	return obj, nil
 }
 
-type configNodeV1 struct{}
+type configNodeV1 struct {
+	nodeLister                   func() corev1listers.NodeLister
```

Comment:
why not to expose `SetExternalKubeInformerFactory` directly on `configNodeV1`, and use it as:
```
func (c *configNodeV1) SetExternalKubeInformerFactory(kubeInformers informers.SharedInformerFactory) {
	c.nodeLister = kubeInformers.Core().V1().Nodes().Lister()
	c.SetReadyFunc(kubeInformers.Core().V1().Nodes().Informer().HasSynced)
}
```
?

Reply:
I'm not sure this pattern was done in https://github.com/openshift/kubernetes/blob/master/openshift-kube-apiserver/admission/customresourcevalidation/route/validation_wrapper.go though isn't done everywhere. I'm not sure why I would or wouldn't. I could drop it